vms/avm/wallet_service.go

```diff
@@ -28,7 +28,36 @@ type WalletService struct {
 }
 
 func (w *WalletService) decided(txID ids.ID) {
+	if _, ok := w.pendingTxs.Get(txID); !ok {
+		return
+	}
+
+	w.vm.ctx.Log.Info("tx decided over wallet API",
+		zap.Stringer("txID", txID),
+	)
 	w.pendingTxs.Delete(txID)
+
+	for {
```

Comment:
Can we document or just help me understand, why pop off oldest transaction only once, and exit the for-loop on success? Trying to understand how `decided` function relates to this operation.

Reply:
To rephrase my question,

> Allow dependent tx issuance

In what cases, dependent tx failed to be issued, so we need another tx issuance to trigger the tx issue

Reply:
If the user is issuing virtuous txs - verification will never fail. However, I didn't want to just have the node crash (or fall into an infinite loop) in this situation... So instead I dropped the tx and logged a warning.

In all likelihood - if one tx is dropped they all will be.

Reply:
> ... didn't want .. or fall into an infinite loop

Oh makes sense.